File: store-realm/src/main/java/packi/day/store/feature/realm/RealmInternationalDay.java
package packi.day.store.feature.realm;

import org.jetbrains.annotations.NotNull;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;
import packi.day.store.CelebrationAdaptable;
import packi.day.store.InternationalDay;

/**
 * Created by lucas34990 on 12/2/18.
 */

public class RealmInternationalDay extends RealmObject implements CelebrationAdaptable {

    @PrimaryKey
    int id;

    String name;

    int day;

    int month;

    String image;

    @NotNull
    @Override
    public InternationalDay adapt() {
        return new InternationalDay(id, name, day, month, image);
    }
}

File: store-realm/src/main/java/packi/day/store/feature/realm/RealmInternationalDay.kt
package packi.day.store.feature.realm

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import org.joda.time.MonthDay
import packi.day.store.Celebration
import packi.day.store.InternationalDay

/**
 * As of Realm 4.3.0
 * Data class is not supported
 * Class has to be open for the proxy generation
 * 👎 for Realm
 */
open class RealmInternationalDay : RealmObject() {

    @PrimaryKey
    var id: Int = 0

    var name: String? = null

    var day: Int = 0

    var month: Int = 0

    var image: String? = null

}

fun RealmInternationalDay.toInternationalDay(): InternationalDay {
    return InternationalDay(MonthDay(month, day), Celebration(id, name.orEmpty(), image))
}